lib/origin_simulator.ex

```diff
@@ -67,6 +67,7 @@ defmodule OriginSimulator do
 
     conn
     |> put_resp_content_type(content_type(body))
+    |> merge_resp_headers(Simulation.recipe(:simulation).headers)
```

Comment:
It looks like this has introduced a bug. @boonious has found this and I replicated.

```
11:50:42.010 [error] #PID<0.1310.0> running OriginSimulator (connection #PID<0.1308.0>, stream id 1) terminated
Server: 127.0.0.1:8080 (http)
Request: GET /data
** (exit) an exception was raised:
    ** (UndefinedFunctionError) function nil.headers/0 is undefined
        nil.headers()
        (origin_simulator) lib/origin_simulator.ex:70: OriginSimulator.serve_payload/1
        (origin_simulator) lib/origin_simulator.ex:1: OriginSimulator.plug_builder_call/2
        (plug_cowboy) lib/plug/cowboy/handler.ex:12: Plug.Cowboy.Handler.init/2
        (cowboy) /Users/FrencS03/Development/origin_simulator/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /Users/FrencS03/Development/origin_simulator/deps/cowboy/src/cowboy_stream_h.erl:296: :cowboy_stream_h.execute/3
        (cowboy) /Users/FrencS03/Development/origin_simulator/deps/cowboy/src/cowboy_stream_h.erl:274: :cowboy_stream_h.request_process/3
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
```

@boonious will look at fixing it.